fix(openpipeline-v1): Align with spec

Kirdock opened the pull request:
#### **Why** this PR?
Several fields were missing, and some were incorrect.
For the Technology processor, `description` and `matcher` can't be set, but `custom_matcher` can.

Fixes #782 

#### **What** has changed?
- Technology processor added (added to span example)
- Cost allocation added
- Product allocation added
- Azure log forwarding processor added
- Security event extraction processor added

#### **How** does it do it?
By adding all missing fields and fixing existing ones.

#### How is it **tested**?
Tests are enabled again, and examples are extended to use all the fields if possible.
Verified locally that the deployed events, spans, and logs example equals the exported config.

#### How does it affect **users**?
Users can now enter everything that is supported by the API spec (API spec and actual allowed values differ; e.g., technology processor is supported by logs but not for events)

**Issue:** CA-16528

### Open question
Do we want to remove the bucket dependency for openpipeline?
The thing is that this one can easily replace unintended fields. If a bucket has the name `custom` and I'm using this in descriptions, the string `custom` is replaced with a bucket reference. For strings/content, this isn't a problem, but for fields like `custom_matcher`.

Kirdock committed:
fix(openpipeline-v1): Align with spec

- Technology processor added (added to span example)
- Cost allocation added
- Product allocation added
- Azure log forwarding processor added
- Security event extraction processor added

Kirdock commented on dynatrace/api/openpipeline/testdata/terraform/example-security-events.tf.disabled:
I renamed and disabled this one for tests as this runs into `Configuration disabled`

arthurpitman commented on dynatrace/testing/api/settingstest.go:
```diff
@@ -230,7 +230,7 @@ func TestAcc(t *testing.T, opts ...TestAccOptions) {
 			}
 			for _, entry := range entries {
-				if !strings.HasSuffix(entry.Name(), ".tf") {
+				if !strings.HasSuffix(entry.Name(), ".tf") || strings.HasSuffix(entry.Name(), "__providers__.tf") {
```
Should this be in its own commit?

arthurpitman commented on dynatrace/api/openpipeline/settings/processor_endpoint.go:
```diff
@@ -69,6 +69,7 @@ type EndpointProcessor struct {
 	fieldsAddProcessor    *FieldsAddProcessor
 	fieldsRemoveProcessor *FieldsRemoveProcessor
 	fieldsRenameProcessor *FieldsRenameProcessor
+	technologyProcessor   *TechnologyProcessor
```
nit: should we make this line last in the struct?

Kirdock replied:
I added it here so it's aligned with the spec
<img width="448" height="188" alt="image" src="https://github.com/user-attachments/assets/1055b2fb-f36c-40b4-bbe9-19af69ae7e7d" />

Kirdock committed:
test: exclude provider files in tests

For manually applying configs, a provider file is needed, and this one should be ignored for testing
